### ISSUE TITLE
Nested class in Protocol treated as instance variable instead of class variable

### ISSUE THREAD

**Issue by Kazy** [microsoft/pyright#3059]
When defining a protocol expecting a nested class to be defined, defining the nested class in a subclass with Pyright in strict mode raises an error: `Instance variable "Config" overrides class variable of same name in class "Base"`. 

Here is a reproduction, with the workaround I'm currently using. Apparently doing a variable assignation correctly considers the variable to be a class var, while defining the class itself directly does not.
```python
import typing as t

class ConfigBase:
    ...

class Base(t.Protocol):
    Config: t.ClassVar[t.Type[ConfigBase]]

# Instance variable "Config" overrides class variable of same name in class "Base" Pyright
# (reportIncompatibleVariableOverride)
class Foo(Base):
    class Config(ConfigBase):
        ...

# Typecheck correctly
class Bar(Base):
    class MyConfig(ConfigBase):
        ...

    Config = MyConfig
```

and for the `pyrightconfig.json`:
```json
{
  "typeCheckingMode": "strict",
}
```

Thanks !

**Comment by erictraut** [comment]
Mypy appears to accept this, so I decided to follow suit in pyright. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.223, which I just published. It will also be included in the next release of pylance.

**Comment by Kazy** [comment]
Thanks a lot !
